<a href="https://colab.research.google.com/github/InstitutodaVinci3/Vancouver/blob/master/MedMepNewData.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Bibliotecas Pandas, Numpy e demais necessárias ao trabalho

In [1]:
# importando as bibliotecas básicas e preparando ambiente
import pandas as pd                     # convenção 
import numpy as np                      # convenção 
import random as rd                     # rd não é alias padrão mas sim adotado aqui somente 
# Itens adicionais
import glob
import os
import re

#Pré-requisto para leitura dos arquivos
- para leitura dos arquivos diretamente em XLSX é necessário ter os arquivos dispolibilizados no seu Google Dreive, uma vez que ñ possível carregá-los no GitHub
- Para que não precise alterar o código abaixo crie uma pasta no seu Google Drive com o seguinte caminho / nome **MedMep/New_Data_XLSX** e carregue os arquivos TEMPERATURA_VAGINAL, CONSUMO, GLICOSE_UREIA_INSULINA, LEITE_COMPOSIÇÃO, e Datalogger (todos com a extensão xlsx)
- o caminho final deve ser assim: /content/drive/MyDrive/MedMep/New_Data_ XLSX/*NomeDoArquivo*

In [ ]:
# Pode ser necessário montar o Google Drive se vc ainda não estiver conectado ao mesmo

from google.colab import drive
drive.mount('/content/drive')

# Leitura dos novos arquivos recebidos

**Nomemclatura:**
- arquivo de temperatura vaginal - Data Frame = tv
- arquivo de consumo - Data Frame = cons
- arquivo de glicose e uréia - Data Frame = glu
- arquivo composição do leite - Data Frame = comp
- arquivo Datalogger - Data Frame = dtlog
- arquivo Produção Diária Unificado - Data Frame = prod_dia

##Passo 1 - Lendo o arquivo de temperatura vaginal

In [ ]:
tv = pd.read_excel('/content/drive/MyDrive/MedMep/New_Data_ XLSX/TEMPERATURA_VAGINAL.xlsx')

In [5]:
tv.head(25)

FASE       DATA   DIA NO EXPERIMENTO  ...  PERÍODO      HORA      TV
0      1 2017-07-20                   12  ...        0  07:30:00     NaN
1      1 2017-07-20                   12  ...        0  08:00:00     NaN
2      1 2017-07-20                   12  ...        0  08:30:00     NaN
3      1 2017-07-20                   12  ...        0  09:00:00     NaN
4      1 2017-07-20                   12  ...        0  09:30:00     NaN
5      1 2017-07-20                   12  ...        0  10:00:00     NaN
6      1 2017-07-20                   12  ...        0  10:30:00     NaN
7      1 2017-07-20                   12  ...        0  11:00:00     NaN
8      1 2017-07-20                   12  ...        0  11:30:00  39.125
9      1 2017-07-20                   12  ...        0  12:00:00  39.250
10     1 2017-07-20                   12  ...        0  12:30:00  39.375
11     1 2017-07-20                   12  ...        0  13:00:00  39.625
12     1 2017-07-20                   12  ...        0  13:30:00  39.500
13     1 2017-07-20                   12  ...        0  14:00:00  39.375
14     1 2017-07-20                   12  ...        0  14:30:00  39.500
15     1 2017-07-20                   12  ...        0  15:00:00  39.625
16     1 2017-07-20                   12  ...        0  15:30:00  39.500
17     1 2017-07-20                   12  ...        0  16:00:00  39.375
18     1 2017-07-20                   12  ...        0  16:30:00  40.000
19     1 2017-07-20                   12  ...        0  17:00:00  40.250
20     1 2017-07-20                   12  ...        0  17:30:00  40.000
21     1 2017-07-20                   12  ...        0  18:00:00  39.500
22     1 2017-07-20                   12  ...        0  18:30:00  39.500
23     1 2017-07-20                   12  ...        0  19:00:00  39.500
24     1 2017-07-20                   12  ...        0  19:30:00  39.750

[25 rows x 9 columns]

In [7]:
tv.tail(50)

FASE       DATA  ...                 HORA      TV
29002     2 2017-09-26  ...  1900-01-11 15:00:00  39.125
29003     2 2017-09-26  ...  1900-01-11 15:30:00  39.125
29004     2 2017-09-26  ...  1900-01-11 16:00:00  39.125
29005     2 2017-09-26  ...  1900-01-11 16:30:00  38.750
29006     2 2017-09-26  ...  1900-01-11 17:00:00  38.875
29007     2 2017-09-26  ...  1900-01-11 17:30:00  39.000
29008     2 2017-09-26  ...  1900-01-11 18:00:00  38.375
29009     2 2017-09-26  ...  1900-01-11 18:30:00  34.875
29010     2 2017-09-26  ...  1900-01-11 19:00:00  38.875
29011     2 2017-09-26  ...  1900-01-11 19:30:00  38.750
29012     2 2017-09-26  ...  1900-01-11 20:00:00  37.875
29013     2 2017-09-26  ...  1900-01-11 20:30:00  37.250
29014     2 2017-09-26  ...  1900-01-11 21:00:00  38.750
29015     2 2017-09-26  ...  1900-01-11 21:30:00  37.875
29016     2 2017-09-26  ...  1900-01-11 22:00:00  37.250
29017     2 2017-09-26  ...  1900-01-11 22:30:00     NaN
29018     2 2017-09-26  ...  1900-01-11 23:00:00     NaN
29019     2 2017-09-26  ...  1900-01-11 23:30:00     NaN
29020     2 2017-09-26  ...  1900-01-12 00:00:00     NaN
29021     2 2017-09-27  ...  1900-01-11 00:30:00     NaN
29022     2 2017-09-27  ...  1900-01-11 01:00:00     NaN
29023     2 2017-09-27  ...  1900-01-11 01:30:00     NaN
29024     2 2017-09-27  ...  1900-01-11 02:00:00     NaN
29025     2 2017-09-27  ...  1900-01-11 02:30:00     NaN
29026     2 2017-09-27  ...  1900-01-11 03:00:00     NaN
29027     2 2017-09-27  ...  1900-01-11 03:30:00     NaN
29028     2 2017-09-27  ...  1900-01-11 04:00:00     NaN
29029     2 2017-09-27  ...  1900-01-11 04:30:00     NaN
29030     2 2017-09-27  ...  1900-01-11 05:00:00     NaN
29031     2 2017-09-27  ...  1900-01-11 05:30:00     NaN
29032     2 2017-09-27  ...  1900-01-11 06:00:00     NaN
29033     2 2017-09-27  ...  1900-01-11 06:30:00     NaN
29034     2 2017-09-27  ...  1900-01-11 07:00:00     NaN
29035     2 2017-09-27  ...  1900-01-11 07:30:00     NaN
29036     2 2017-09-27  ...  1900-01-11 08:00:00     NaN
29037     2 2017-09-27  ...  1900-01-11 08:30:00     NaN
29038     2 2017-09-27  ...  1900-01-11 09:00:00     NaN
29039     2 2017-09-27  ...  1900-01-11 09:30:00     NaN
29040     2 2017-09-27  ...  1900-01-11 10:00:00     NaN
29041     2 2017-09-27  ...  1900-01-11 10:30:00     NaN
29042     2 2017-09-27  ...  1900-01-11 11:00:00     NaN
29043     2 2017-09-27  ...  1900-01-11 11:30:00     NaN
29044     2 2017-09-27  ...  1900-01-11 12:00:00     NaN
29045     2 2017-09-27  ...  1900-01-11 12:30:00     NaN
29046     2 2017-09-27  ...  1900-01-11 13:00:00     NaN
29047     2 2017-09-27  ...  1900-01-11 13:30:00     NaN
29048     2 2017-09-27  ...  1900-01-11 14:00:00     NaN
29049     2 2017-09-27  ...  1900-01-11 14:30:00     NaN
29050     2 2017-09-27  ...  1900-01-11 15:00:00     NaN
29051     2 2017-09-27  ...  1900-01-11 15:30:00     NaN

[50 rows x 9 columns]

In [8]:
tv.describe()

FASE   DIA NO EXPERIMENTO  ...       PERÍODO            TV
count  29052.000000         29052.000000  ...  29052.000000  26051.000000
mean       1.458488            19.793061  ...      0.897150     38.819522
std        0.498282             5.081999  ...      0.303768      2.755867
min        1.000000            11.000000  ...      0.000000      0.000000
25%        1.000000            15.000000  ...      1.000000     38.750000
50%        1.000000            20.000000  ...      1.000000     39.000000
75%        2.000000            24.000000  ...      1.000000     39.250000
max        2.000000            30.000000  ...      1.000000     42.125000

[8 rows x 7 columns]

##Passo 2 - Lendo o arquivo de consumo

In [9]:
cons = pd.read_excel('/content/drive/MyDrive/MedMep/New_Data_ XLSX/CONSUMO.xls')

In [10]:
cons.head(25)

FASE       DATA  DIA  ANIMAL  DIETA  ...  PER  OFER  SOBRA     MN       CMS
0      2 2017-09-01    0     653      1  ...    0  53.0   1.70  51.30  20.77196
1      2 2017-09-02    1     653      1  ...    0  55.0   0.44  54.56  22.09197
2      2 2017-09-03    2     653      1  ...    0  58.0   1.92  56.08  22.70744
3      2 2017-09-04    3     653      1  ...    0  60.0   3.38  56.62  22.92609
4      2 2017-09-05    4     653      1  ...    0  60.0   8.44  51.56  20.87724
5      2 2017-09-06    5     653      1  ...    0  58.0   4.00  54.00  21.86522
6      2 2017-09-07    6     653      1  ...    0  58.0   4.75  53.25  21.56154
7      2 2017-09-08    7     653      1  ...    0  58.0   2.60  55.40  22.07462
8      2 2017-09-09    8     653      1  ...    0  60.0  16.88  43.12  17.18154
9      2 2017-09-10    9     653      1  ...    0  62.0   0.00  62.00  24.70445
10     2 2017-09-11   10     653      1  ...    0  66.0  52.92  13.08       NaN
11     2 2017-09-12   11     653      1  ...    0  50.0   3.04  46.96  18.71163
12     2 2017-09-13   12     653      1  ...    0  52.0   0.00  52.00  20.71986
13     2 2017-09-14   13     653      1  ...    0  58.0   6.53  51.47  20.50868
14     2 2017-09-15   14     653      1  ...    1  58.0   0.58  57.42  19.94779
15     2 2017-09-16   15     653      1  ...    1  64.0   6.68  57.32  19.91305
16     2 2017-09-17   16     653      1  ...    1  64.0  14.36  49.64  17.24501
17     2 2017-09-18   17     653      1  ...    1  60.0  11.00  49.00  17.02267
18     2 2017-09-19   18     653      1  ...    1  55.0   6.58  48.42  16.82118
19     2 2017-09-20   19     653      1  ...    1  55.0   1.22  53.78  18.68325
20     2 2017-09-21   20     653      1  ...    1  60.0  10.18  49.82  17.30754
21     2 2017-09-22   21     653      1  ...    1  58.0   0.60  57.40  20.58011
22     2 2017-09-23   22     653      1  ...    1  64.0   7.34  56.66  20.31479
23     2 2017-09-24   23     653      1  ...    1  64.0   1.46  62.54  22.42299
24     2 2017-09-25   24     653      1  ...    1  70.0  17.38  52.62  18.86629

[25 rows x 11 columns]

In [11]:
cons.tail(25)

FASE       DATA  DIA  ANIMAL  DIETA  ...  PER  OFER  SOBRA     MN        CMS
983      2 2017-09-02    1    4498      1  ...    0  55.0   3.50  51.50  20.401720
984      2 2017-09-03    2    4498      1  ...    0  56.0   4.26  51.74  20.496800
985      2 2017-09-04    3    4498      1  ...    0  55.0   3.66  51.34  20.338340
986      2 2017-09-05    4    4498      1  ...    0  55.0   7.00  48.00  19.015200
987      2 2017-09-06    5    4498      1  ...    0  53.0   6.40  46.60  18.460590
988      2 2017-09-07    6    4498      1  ...    0  52.0  21.70  30.30  12.003340
989      2 2017-09-08    7    4498      1  ...    0  50.0  11.40  38.60  14.967220
990      2 2017-09-09    8    4498      1  ...    0  50.0  11.60  38.40  14.889670
991      2 2017-09-10    9    4498      1  ...    0  42.0   0.00  42.00  16.285570
992      2 2017-09-11   10    4498      1  ...    0  50.0   0.00  50.00  19.387590
993      2 2017-09-12   11    4498      1  ...    0  54.0   1.00  53.00  20.550840
994      2 2017-09-13   12    4498      1  ...    0  57.0  25.86  31.14        NaN
995      2 2017-09-14   13    4498      1  ...    0  50.0  27.36  22.64        NaN
996      2 2017-09-15   14    4498      1  ...    1  50.0  19.80  30.20   9.853070
997      2 2017-09-16   15    4498      1  ...    1  50.0  22.76  27.24   8.887338
998      2 2017-09-17   16    4498      1  ...    1  50.0   0.00  50.00  16.313030
999      2 2017-09-18   17    4498      1  ...    1  55.0   7.28  47.72  15.569150
1000     2 2017-09-19   18    4498      1  ...    1  55.0   3.00  52.00  16.965550
1001     2 2017-09-20   19    4498      1  ...    1  57.0   3.90  53.10  17.324440
1002     2 2017-09-21   20    4498      1  ...    1  59.0   4.84  54.16  17.670270
1003     2 2017-09-22   21    4498      1  ...    1  60.0  14.95  45.05  15.279590
1004     2 2017-09-23   22    4498      1  ...    1  55.0   1.92  53.08  18.003120
1005     2 2017-09-24   23    4498      1  ...    1  60.0   0.86  59.14  20.058490
1006     2 2017-09-25   24    4498      1  ...    1  67.0   9.34  57.66  19.556510
1007     2 2017-09-26   25    4498      1  ...    1  66.0   8.75  57.25  19.417450

[25 rows x 11 columns]

In [12]:
cons.describe()

FASE          DIA  ...           MN         CMS
count  1008.000000  1008.000000  ...  1006.000000  999.000000
mean      1.464286    13.571429  ...    49.277515   19.139541
std       0.498970     8.204369  ...     8.837200    2.785949
min       1.000000     0.000000  ...    10.280000    0.000000
25%       1.000000     6.750000  ...    43.325000   17.428820
50%       1.000000    13.500000  ...    48.580000   19.191250
75%       2.000000    20.250000  ...    54.835000   20.954445
max       2.000000    29.000000  ...    76.200000   27.173630

[8 rows x 10 columns]

##Passo 3 - Lendo o arquivo de Glicose, Uréia e Insulina 

In [13]:
glu = pd.read_excel('/content/drive/MyDrive/MedMep/New_Data_ XLSX/GLICOSE_UREIA_INSULINA.xlsx')

In [14]:
glu.head(25)

FASE       DATA   DIA NO EXPERIMENTO  ...  UREIA  GLICOSE  INSULINA
0      1 2017-07-09                    1  ...     16       53     20.11
1      1 2017-07-09                    1  ...     23       64     19.96
2      1 2017-07-09                    1  ...     15       63     30.46
3      1 2017-07-09                    1  ...     17       51     10.98
4      1 2017-07-09                    1  ...     16       61       NaN
5      1 2017-07-09                    1  ...     12       60     36.42
6      1 2017-07-09                    1  ...     18       54     20.13
7      1 2017-07-09                    1  ...     16       49     22.75
8      1 2017-07-09                    1  ...     18       47     63.08
9      1 2017-07-09                    1  ...     17       62     26.96
10     1 2017-07-09                    1  ...     19       56     20.70
11     1 2017-07-09                    1  ...     15       43     16.98
12     1 2017-07-09                    1  ...     22       63     36.37
13     1 2017-07-09                    1  ...     16       53     21.56
14     1 2017-07-09                    1  ...     13       56     42.69
15     1 2017-07-09                    1  ...     20       56     21.07
16     1 2017-07-09                    1  ...     18       59     28.14
17     1 2017-07-09                    1  ...     15       58     32.24
18     1 2017-07-19                   11  ...     22       64     13.25
19     1 2017-07-19                   11  ...     27       76     25.02
20     1 2017-07-19                   11  ...     20       77     49.28
21     1 2017-07-19                   11  ...     23       84     19.84
22     1 2017-07-19                   11  ...     21       75     37.85
23     1 2017-07-19                   11  ...     17       73     14.50
24     1 2017-07-19                   11  ...     22       69     16.31

[25 rows x 10 columns]

In [15]:
glu.tail(25)

FASE       DATA   DIA NO EXPERIMENTO  ...  UREIA  GLICOSE  INSULINA
155     2 2017-09-22                   21  ...     22       61      5.94
156     2 2017-09-22                   21  ...     27       70     10.76
157     2 2017-09-22                   21  ...     25       75     31.02
158     2 2017-09-22                   21  ...     22       60     17.41
159     2 2017-09-22                   21  ...     28       49      4.98
160     2 2017-09-22                   21  ...     33       61     34.23
161     2 2017-09-22                   21  ...     36       61     18.00
162     2 2017-09-26                   25  ...     23       58     33.64
163     2 2017-09-26                   25  ...     21       67     14.11
164     2 2017-09-26                   25  ...     25       64     26.39
165     2 2017-09-26                   25  ...     25       71     39.13
166     2 2017-09-26                   25  ...     18       64      9.99
167     2 2017-09-26                   25  ...     16       65     51.85
168     2 2017-09-26                   25  ...     20       68     15.39
169     2 2017-09-26                   25  ...     25       93     72.13
170     2 2017-09-26                   25  ...     25       69     22.84
171     2 2017-09-27                   26  ...     23       41     17.89
172     2 2017-09-27                   26  ...     16       74     25.70
173     2 2017-09-27                   26  ...     29       57     17.51
174     2 2017-09-27                   26  ...     21       64     13.64
175     2 2017-09-27                   26  ...     23       79     26.70
176     2 2017-09-27                   26  ...     21       59     14.84
177     2 2017-09-27                   26  ...     25       60      7.34
178     2 2017-09-27                   26  ...     21       69     21.32
179     2 2017-09-27                   26  ...     30       61     10.60

[25 rows x 10 columns]

In [17]:
glu.describe(percentiles=None)

FASE   DIA NO EXPERIMENTO  ...     GLICOSE    INSULINA
count  180.000000           180.000000  ...  180.000000  179.000000
mean     1.500000            15.450000  ...   64.844444   21.323408
std      0.501395             9.027751  ...    7.616376   13.841551
min      1.000000             1.000000  ...   41.000000    1.950000
25%      1.000000            11.000000  ...   60.750000   11.560000
50%      1.500000            17.000000  ...   65.000000   17.890000
75%      2.000000            20.250000  ...   69.250000   26.940000
max      2.000000            30.000000  ...   93.000000   99.410000

[8 rows x 9 columns]

##Passo 4 - Lendo o arquivo de Composição do Leite

In [26]:
comp = pd.read_excel('/content/drive/MyDrive/MedMep/New_Data_ XLSX/LEITE_COMPOSIÇÃO.xlsx')

In [27]:
comp.head(25)

FASE       DATA  DIA NO EXPERIMENTO  ...        ECM        FCM  PRODGORD
0      1 2017-07-11                   3  ...  29.611674  33.331680   1.28160
1      1 2017-07-11                   3  ...        NaN        NaN       NaN
2      1 2017-07-12                   4  ...  27.165004  30.273705   1.14435
3      1 2017-07-12                   4  ...        NaN        NaN       NaN
4      1 2017-07-18                  10  ...  27.162655  29.144027   1.03529
5      1 2017-07-18                  10  ...        NaN        NaN       NaN
6      1 2017-07-19                  11  ...  27.209894  29.991918   1.10586
7      1 2017-07-19                  11  ...        NaN        NaN       NaN
8      1 2017-07-25                  17  ...  30.621679  34.399355   1.33385
9      1 2017-07-25                  17  ...        NaN        NaN       NaN
10     1 2017-07-26                  18  ...  28.666040  31.860595   1.19665
11     1 2017-07-26                  18  ...        NaN        NaN       NaN
12     1 2017-07-29                  21  ...  29.544353  32.938617   1.29459
13     1 2017-07-29                  21  ...        NaN        NaN       NaN
14     1 2017-07-30                  22  ...  28.993992  32.261969   1.27163
15     1 2017-07-30                  22  ...        NaN        NaN       NaN
16     1 2017-08-01                  24  ...        NaN        NaN       NaN
17     1 2017-08-01                  24  ...        NaN        NaN       NaN
18     1 2017-08-02                  25  ...  29.008534  32.416879   1.24933
19     1 2017-08-02                  25  ...        NaN        NaN       NaN
20     1 2017-08-05                  28  ...  28.761140  31.978735   1.22245
21     1 2017-08-05                  28  ...        NaN        NaN       NaN
22     1 2017-08-06                  29  ...  29.453694  32.997139   1.29553
23     1 2017-08-06                  29  ...        NaN        NaN       NaN
24     1 2017-07-11                   3  ...  23.903284  26.309696   0.99392

[25 rows x 24 columns]

In [28]:
comp.tail(25)

FASE       DATA  DIA NO EXPERIMENTO  ...        ECM        FCM  PRODGORD
767     2 2017-09-24                  23  ...        NaN        NaN       NaN
768     2 2017-09-26                  25  ...  21.517160  23.632765   0.90655
769     2 2017-09-26                  25  ...        NaN        NaN       NaN
770     2 2017-09-27                  26  ...  21.628653  24.609050   1.01150
771     2 2017-09-27                  26  ...        NaN        NaN       NaN
772     2 2017-09-05                   4  ...  21.397560  23.576166   0.83682
773     2 2017-09-05                   4  ...        NaN        NaN       NaN
774     2 2017-09-06                   5  ...  18.186756  19.637304   0.66408
775     2 2017-09-06                   5  ...        NaN        NaN       NaN
776     2 2017-09-12                  11  ...  17.265264  19.074425   0.65075
777     2 2017-09-12                  11  ...        NaN        NaN       NaN
778     2 2017-09-13                  12  ...  14.874319  16.445403   0.48681
779     2 2017-09-13                  12  ...        NaN        NaN       NaN
780     2 2017-09-19                  18  ...  20.258639  22.246831   0.79237
781     2 2017-09-19                  18  ...        NaN        NaN       NaN
782     2 2017-09-20                  19  ...  19.776375  21.489202   0.74854
783     2 2017-09-20                  19  ...        NaN        NaN       NaN
784     2 2017-09-23                  22  ...  20.086791  22.089862   0.78274
785     2 2017-09-23                  22  ...        NaN        NaN       NaN
786     2 2017-09-24                  23  ...  20.226042  22.330149   0.78423
787     2 2017-09-24                  23  ...        NaN        NaN       NaN
788     2 2017-09-26                  25  ...  20.841710  22.810703   0.79781
789     2 2017-09-26                  25  ...        NaN        NaN       NaN
790     2 2017-09-27                  26  ...  18.148899  19.615344   0.63888
791     2 2017-09-27                  26  ...        NaN        NaN       NaN

[25 rows x 24 columns]

In [29]:
comp.describe()

FASE  DIA NO EXPERIMENTO  ...         FCM    PRODGORD
count  792.000000          792.000000  ...  385.000000  385.000000
mean     1.454545           17.136364  ...   24.191769    0.933479
std      0.498244            8.136422  ...    5.163208    0.241832
min      1.000000            3.000000  ...    9.073035    0.283560
25%      1.000000           11.000000  ...   20.507223    0.787700
50%      1.000000           18.500000  ...   24.580272    0.942820
75%      2.000000           24.000000  ...   27.181756    1.063370
max      2.000000           29.000000  ...   46.715802    2.346540

[8 rows x 14 columns]

##Passo 5 - Lendo o arquivo de Datalogger

**Necessário rever este passo pois o arquivo Datalogger possui múltiplas Tabs**

In [22]:
dtlog = pd.read_excel('/content/drive/MyDrive/MedMep/New_Data_ XLSX/Datalogger.xls')

In [23]:
dtlog.head(25)

NO.        DATE      TIME  ...  RELATIVE-HUMIDITY  DEW-POINT        ITU
0     1  07-21-2017  07:04:17  ...               82.7       10.6  56.461755
1     2  07-21-2017  07:34:17  ...               87.7       11.4  56.247182
2     3  07-21-2017  08:04:17  ...               88.3       12.3  57.756731
3     4  07-21-2017  08:34:17  ...               88.0       13.4  59.606480
4     5  07-21-2017  09:04:17  ...               85.8       14.1  61.411030
5     6  07-21-2017  09:34:17  ...               80.9       14.8  64.049862
6     7  07-21-2017  10:04:17  ...               79.4       15.1  64.951728
7     8  07-21-2017  10:34:17  ...               76.3       15.4  66.226889
8     9  07-21-2017  11:04:17  ...               74.0       15.6  67.187040
9    10  07-21-2017  11:34:17  ...               71.5       15.9  68.405705
10   11  07-21-2017  12:04:17  ...               70.6       16.4  69.400880
11   12  07-21-2017  12:34:17  ...               68.5       16.4  69.987875
12   13  07-21-2017  13:04:17  ...               64.9       16.3  70.862783
13   14  07-21-2017  13:34:17  ...               60.6       16.0  71.613754
14   15  07-21-2017  14:04:17  ...               55.7       15.7  72.642936
15   16  07-21-2017  14:34:17  ...               54.7       16.2  73.752833
16   17  07-21-2017  15:04:17  ...               52.5       15.9  74.030875
17   18  07-21-2017  15:34:17  ...               51.3       15.7  74.019442
18   19  07-21-2017  16:04:17  ...               48.1       15.8  75.177782
19   20  07-21-2017  16:34:17  ...               53.4       16.0  73.870558
20   21  07-21-2017  17:04:17  ...               66.5       15.2  68.800530
21   22  07-21-2017  17:34:17  ...               70.5       14.5  66.528295
22   23  07-21-2017  18:04:17  ...               74.4       15.0  66.128032
23   24  07-21-2017  18:34:17  ...               78.7       15.1  65.080457
24   25  07-21-2017  19:04:17  ...               79.0       14.9  64.775270

[25 rows x 7 columns]

In [24]:
dtlog.tail(25)

NO.                 DATE  ... DEW-POINT        ITU
825  826  2017-07-08 00:00:00  ...      13.6  66.795755
826  827  2017-07-08 00:00:00  ...      13.8  67.665376
827  828  2017-07-08 00:00:00  ...      13.9  68.414584
828  829  2017-07-08 00:00:00  ...      14.2  69.500897
829  830  2017-07-08 00:00:00  ...      13.9  70.064986
830  831  2017-07-08 00:00:00  ...      14.0  70.518324
831  832  2017-07-08 00:00:00  ...      13.9  70.685857
832  833  2017-07-08 00:00:00  ...      14.9  72.068891
833  834  2017-07-08 00:00:00  ...      14.3  72.042637
834  835  2017-07-08 00:00:00  ...      14.2  71.810116
835  836  2017-07-08 00:00:00  ...      14.4  72.185072
836  837  2017-07-08 00:00:00  ...      14.0  68.188080
837  838  2017-07-08 00:00:00  ...      13.6  66.416982
838  839  2017-07-08 00:00:00  ...      14.0  65.107916
839  840  2017-07-08 00:00:00  ...      13.2  62.777025
840  841  2017-07-08 00:00:00  ...      12.6  61.256370
841  842  2017-07-08 00:00:00  ...      12.6  60.189046
842  843  2017-07-08 00:00:00  ...      12.5  59.402786
843  844  2017-07-08 00:00:00  ...      12.1  58.745585
844  845  2017-07-08 00:00:00  ...      11.5  57.597994
845  846  2017-07-08 00:00:00  ...      11.6  57.093843
846  847  2017-07-08 00:00:00  ...      10.9  56.102019
847  848  2017-07-08 00:00:00  ...      10.6  55.258152
848  849  2017-07-08 00:00:00  ...      10.2  54.415990
849  850  2017-07-08 00:00:00  ...      10.3  54.581096

[25 rows x 7 columns]

In [25]:
dtlog.describe()

NO.  TEMPERATURE  RELATIVE-HUMIDITY   DEW-POINT         ITU
count  850.00000   850.000000         850.000000  850.000000  850.000000
mean   425.50000    17.886118          74.996588   12.938588   62.723641
std    245.51816     4.605151          14.871563    2.000581    6.436493
min      1.00000     8.000000          40.300000    6.400000   47.050760
25%    213.25000    14.025000          62.325000   11.625000   57.301405
50%    425.50000    17.500000          80.900000   13.100000   62.823383
75%    637.75000    21.600000          87.900000   14.500000   68.340043
max    850.00000    28.500000          93.500000   17.200000   75.685310

#Passo 6 - Carregar o arquivo de Produção Diária Unificado
- Este arquivo foi gerado no trabalho anterior unindo os 177 arquivos TXT enviadps
- link do Git Hub para este trabalho citado acima: https://github.com/InstitutodaVinci3/Vancouver/blob/master/Producao_Diaria.ipynb

In [30]:
prod_dia = pd.read_excel('/content/drive/MyDrive/MedMep/New_Data_ XLSX/Prod_Diaria_Unificado_15-01.xlsx')

In [31]:
prod_dia.head(10)

Unnamed: 0  Grupo  Id_Animal  ...  Prod_S2_ont Prod_S3_ont        Data
0           0      1       5402  ...          NaN         NaN  03/05/2017
1           1      1       5403  ...          NaN         NaN  03/05/2017
2           2      1       5407  ...          NaN         NaN  03/05/2017
3           3      1       5408  ...          NaN         NaN  03/05/2017
4           4      1       5409  ...          NaN         NaN  03/05/2017
5           5      1       4584  ...          NaN         NaN  03/05/2017
6           6      1       4585  ...          NaN         NaN  03/05/2017
7           7      1       4586  ...          NaN         NaN  03/05/2017
8           8      1       4588  ...          NaN         NaN  03/05/2017
9           9      1       4573  ...          NaN         NaN  03/05/2017

[10 rows x 25 columns]

In [32]:
prod_dia.tail(10)

Unnamed: 0  Grupo  ...                 Prod_S3_ont        Data
9989          62      6  ...                              26/10/2017
9990          63      7  ...                              26/10/2017
9991          64      7  ...                              26/10/2017
9992          65      7  ...                              26/10/2017
9993          66      7  ...                              26/10/2017
9994          67      7  ...                              26/10/2017
9995          68      7  ...                              26/10/2017
9996          69      7  ...                              26/10/2017
9997          70      7  ...                              26/10/2017
9998          71      7  ...                              26/10/2017

[10 rows x 25 columns]

In [33]:
prod_dia.describe()

Unnamed: 0        Grupo    Id_Animal      Num_Lac
count  9999.000000  9999.000000  9999.000000  9999.000000
mean     29.969097     3.543054  4145.759376     1.184018
std      19.445019     2.044446   858.051297     0.908023
min       0.000000     1.000000   653.000000     0.000000
25%      14.000000     2.000000  3600.000000     1.000000
50%      28.000000     3.000000  4536.000000     1.000000
75%      45.000000     5.000000  4575.000000     1.000000
max      71.000000     7.000000  5409.000000     8.000000

**Nota:** Necessário verificar porque o describe não está trazendo as estat´sticas para as demais colunas do dataframe

# Passo 7 - Juntar os arquivos novos em um único Data Frame

### A junção dos DataFrames que posseum o **Id de Animal** como chave deve ser priorizado
- tv, cons, glu, comp podem ser concatenados usando **Animal Id, Data e Hora**
- Depois pode-se concatenar com o arquivo Produção DIaria unificado, criado com os dados anteriores 
- Datalogger concatenado por Data e hora ?
